rgapis/client.go

```diff
@@ -0,0 +1,46 @@
+package rgapis
+
+import (
+	"context"
+	"net/http"
+	"strings"
+)
+
+type rgapiClient struct {
+	client     *http.Client
+	region     string
+	rgapiToken string
+}
+
+const RGAPI_TOKEN_HEADER_NAME = "X-Riot-Token"
+const RGAPI_BASE_URI = "https://{region}.api.riotgames.com"
+
+func NewRGAPIClient(rgapiToken string) *rgapiClient {
+	return &rgapiClient{
+		client:     http.DefaultClient,
+		rgapiToken: rgapiToken,
+	}
+}
+
+func (rc *rgapiClient) WithRegion(region string) {
+	rc.region = region
```

Comment:
Looks like a race condition; what if two different requests come in at the same time with different regions and this shared variable has to be overwritten? `DoReq` probably needs to take the region

Reply:
Definitely, the state of the service is being shared between requests so would be a problem.

Region is now a parameter